BritneyS/GetFed
# Add unit tests

## What you did :question:
- Started to add unit tests to project for FoodEntryViewController
- TODO: refactor CoreDataManager to make module more testable

## How you did it :white_check_mark:
- Added a new test target (iOS Unit Testing Bundle)
- Setup view controller in `setup()` method of test module
- Added `ViewControllerID` enum for view controller identifier 
- Started out with testing the successful load of each text field in the Food Entry view
- Notes: These are just preliminary tests! I have future plans to test the insertion of a new Food Entity as well


## How to test it :microscope:
- Open the xcworkspace file of this project in XCode
- Add GetFedTests to GetFed scheme: `GetFed > Edit Scheme... > Test` OR `Product > Scheme > Edit Scheme... > Test`, then click `+` and select `GetFedTests` (see screenshot) //TODO: Config file/setting to make this automatic upon cloning 
- Hit `cmd + u` to run all unit tests
- Note that all unit tests pass
- **If unit tests don't run: Make sure the Deployment Target for the GetFed target is set to at least 12.0**: `GetFed in Project Navigator > GetFed Target > General > Deployment Info`

## Screenshots (if applicable) :camera:
![screen shot 2019-01-16 at 4 09 36 pm](https://user-images.githubusercontent.com/8409475/51278693-43d96980-19a9-11e9-89ca-45f3026ba444.png)

![screen shot 2019-01-16 at 4 21 12 pm](https://user-images.githubusercontent.com/8409475/51279314-08d83580-19ab-11e9-935d-a341e07787dd.png)


![screen shot 2019-01-16 at 11 05 05 am](https://user-images.githubusercontent.com/8409475/51261643-a2d5b900-197e-11e9-886d-67b21a46b402.png)
